backend/dataall/modules/s3_datasets/services/dataset_table_data_filter_service.py

```diff
@@ -0,0 +1,115 @@
+import logging
+import re
+from dataall.base.context import get_context
+from dataall.core.permissions.services.resource_policy_service import ResourcePolicyService
+from dataall.modules.s3_datasets.db.dataset_table_data_filter_repositories import DatasetTableDataFilterRepository
+from dataall.modules.s3_datasets.db.dataset_table_repositories import DatasetTableRepository
+from dataall.modules.s3_datasets.db.dataset_repositories import DatasetRepository
+from dataall.modules.s3_datasets.services.dataset_table_data_filter_enums import DataFilterType
+from dataall.modules.s3_datasets.services.dataset_service import DatasetService
+from dataall.modules.s3_datasets.services.dataset_permissions import (
+    CREATE_TABLE_DATA_FILTER,
+    DELETE_TABLE_DATA_FILTER,
+    LIST_TABLE_DATA_FILTERS,
+)
+from dataall.base.db import exceptions
+from dataall.modules.s3_datasets.aws.lf_data_filter_client import LakeFormationDataFilterClient
+
+
+log = logging.getLogger(__name__)
+
+
+class DatasetTableDataFilterRequestValidationService:
+    @staticmethod
+    def _required_param(param, name):
+        if not param:
+            raise exceptions.RequiredParameter(name)
+
+    @staticmethod
+    def validate_data_filter_type(data):
+        DatasetTableDataFilterRequestValidationService._required_param(data.get('filterType'), 'filterType')
+        if data.get('filterType') not in set(item.value for item in DataFilterType):
+            raise exceptions.InvalidInput(
+                'filterType',
+                data.get('filterType'),
+                'ROW or COLUMN value',
+            )
+        if data.get('filterType') == DataFilterType.ROW.value and not data.get('rowExpression'):
+            raise exceptions.InvalidInput(
+                'rowExpression',
+                data.get('rowExpression'),
+                f'must be provided for {data.get("filterType")} filter',
+            )
+        if data.get('filterType') == DataFilterType.COLUMN.value and not data.get('includedCols'):
+            raise exceptions.InvalidInput(
+                'includedCols',
+                data.get('includedCols'),
+                f'must be provided for {data.get("filterType")} filter',
+            )
+
+    @staticmethod
+    def validate_creation_data_filter_params(uri, data):
+        DatasetTableDataFilterRequestValidationService._required_param(uri, 'tableUri')
+        DatasetTableDataFilterRequestValidationService._required_param(data, 'data')
+        DatasetTableDataFilterRequestValidationService._required_param(data.get('filterName'), 'filterName')
```

Comment:
Does this constraint come from the naming convention of Lake Formation filters? In that case we could use the naming convention utilities no?

Reply:
yes coming from LF / Glue naming convention - will look into using that instead of `re.search()`

Reply:
Using Naming Convention now with a new `validate_name` method